runtime/compiler/control/MongoLogger.cpp

```diff
@@ -0,0 +1,127 @@
+#include "MongoLogger.hpp"
+#include <ctime>
+#include <iostream>
+#include "LoadDBLibs.hpp"
+
+MongoLogger::MongoLogger(std::string const &databaseIP, std::uint32_t databasePort, std::string const &databaseName)
+        : BasePersistentLogger(databaseIP, databasePort, databaseName){
+
+}
+
+MongoLogger::MongoLogger(std::string const &databaseIP, std::uint32_t databasePort, std::string const &databaseName,
+        std::string const &databaseUsername, std::string const &databasePassword)
+        : BasePersistentLogger(databaseIP, databasePort, databaseName, databaseUsername, databasePassword){
+
+}
+
+MongoLogger::~MongoLogger() {
+    //Clean up this mongoc logger.
+    Omongoc_collection_destroy(_collection);
```

Comment:
Is it safe to destroy NULL pointers?

Reply:
Yes it is. The mongoc functions will do nothing if they receive a NULL pointer.